lacework/resource_lacework_team_member.go

```diff
@@ -0,0 +1,557 @@
+							DiffSuppressFunc: diffCaseInsensitive,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								StateFunc: func(val interface{}) string {
+									return strings.ToUpper(strings.TrimSpace(val.(string)))
+								},
+							},
+						},
+					},
+				},
+			},
+			"guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkTeamMemberCreate(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	if lacework.OrgAccess() {
+		return laceworkTeamMemberCreateOrg(d, meta)
+	}
+
+	return laceworkTeamMemberCreate(d, meta)
+}
+
+func laceworkTeamMemberCreateOrg(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	tmOrg := api.NewTeamMemberOrg(d.Get("email").(string),
+		api.TeamMemberProps{
+			AccountAdmin: d.Get("administrator").(bool),
+			Company:      d.Get("company").(string),
+			FirstName:    d.Get("first_name").(string),
+			LastName:     d.Get("last_name").(string),
+		})
+
+	if !d.Get("enabled").(bool) {
+		tmOrg.UserEnabled = 0
+	}
+
+	tmOrg.OrgAdmin = d.Get("organization.0.administrator").(bool)
+	tmOrg.OrgUser = d.Get("organization.0.user").(bool)
+	tmOrg.AdminRoleAccounts = castAndUpperStringSlice(d, "organization.0.admin_accounts")
+	tmOrg.UserRoleAccounts = castAndUpperStringSlice(d, "organization.0.user_accounts")
+
+	if len(tmOrg.AdminRoleAccounts) != 0 || len(tmOrg.UserRoleAccounts) != 0 {
+		// if admin_accounts or user_accounts are set, turn off OrgUser which is turned on by default
+		tmOrg.OrgUser = false
+	}
+
+	if err := validateOrgTeamMember(&tmOrg); err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Creating org team member with data %+v\n", tmOrg)
+	response, err := lacework.V2.TeamMembers.CreateOrg(tmOrg)
+	if err != nil {
+		return err
+	}
+
+	if len(response.Data.Accounts) == 0 {
+		msg := `
+Unable to read sever response data. (empty 'accounts' field)
+
+This was an unexpected behavior, verify that your team member has been
+created successfully and report this issue to support@lacework.net
+`
+		return errors.New(msg)
+	}
+
+	d.SetId(response.Data.Accounts[0].UserGuid)
+	d.Set("guid", response.Data.Accounts[0].UserGuid)
+
+	org := make(map[string]interface{})
+	org["admin_accounts"] = tmOrg.AdminRoleAccounts
+	org["user_accounts"] = tmOrg.UserRoleAccounts
+	org["user"] = tmOrg.OrgUser
+	org["administrator"] = tmOrg.OrgAdmin
+	d.Set("organization", []map[string]interface{}{org})
+
+	log.Printf("[INFO] Created org team member with email %s and guid %s\n",
+		response.Data.UserName, d.Id())
+	return nil
+}
+
+func validateOrgTeamMember(m *api.TeamMemberOrg) error {
```

Comment:
Have you considered shifting this to api/team_members.go

`func (m *TeamMemberOrg)ValidateOrgTeamMember() error {
...
`

And a further thought, making the validate func private and calling it inside the NewTeamMemberOrg constructor?

Reply:
This sounds like a really great idea! Let me create a ticket for it: https://lacework.atlassian.net/browse/ALLY-798